### PR TITLE
Misc. fixes for checkpoint-restart of SysV SHM objects

### PR DESCRIPTION
**Pull request by rohgarg**

### CODE REVIEW

File: src/plugin/svipc/sysvipcwrappers.cpp [hunk 1]
```diff
@@ -68,9 +68,10 @@ shmget(key_t key, size_t size, int shmflg)
   int virtId = -1;
 
   DMTCP_PLUGIN_DISABLE_CKPT();
-  realId = _real_shmget(key, size, shmflg);
+  key_t realKey = dmtcp_virtual_to_real_pid(getpid());
```
Comment (rohgarg) [inline src/plugin/svipc/sysvipcwrappers.cpp]: Thanks for catching this, @balajir1990. I'll fix this in the next iteration.

File: src/plugin/svipc/sysvipc.cpp [hunk 2]
```diff
@@ -430,6 +430,37 @@ SysVIPC::serialize(jalib::JBinarySerializer &o)
...
 }
 
+int
+SysVShm::virtualToRealKey(key_t k)
+{
+  if (_keyMap.find(k) != _keyMap.end()) {
+    return _keyMap[k];
+  } else {
+    int realId = SharedData::getRealIPCId(SYSV_SHM_KEY, k);
+    updateKeyMapping(k, realId);
```
Comment (jiajuncao) [inline src/plugin/svipc/sysvipc.cpp]: Shall we update the mapping only if `realID != -1`? Just to avoid inserting a useless entry.

**Comment by rohgarg** [comment]
Thanks for the feedback, @jiajuncao. I have now updated the patch to address your comments.

**Comment by gc00** [comment]
Hi @rohgarg, This PR has been approved for a while now.  Could you rebase and push it in.  :-)  Thanks.

**Pull request merged**